### PR TITLE
Add MAGIC sensitivity with  references

### PR DESCRIPTION
**Pull request by vuillaut**
Following the discussions about having a proper reference for MAGIC sensitivity, here is a PR fixing that.

- MAGIC sensitivity (lima 5off as default but others available too) with ref from ctaplot
- plot CRAB spectrum from pyIRF
- cleanup and fix a few formatting issues
- add unit tests on changed functions

The result should look like this:
```
import matplotlib.pyplot as plt
import astropy.units as u
from lstchain.mc.plot_utils import plot_Crab_SED
from  ctaplot.plots import plot_sensitivity_magic_performance
plot_Crab_SED(10*u.GeV, 100*u.TeV)
plot_Crab_SED(10*u.GeV, 100*u.TeV, percentage=1, linestyle='--', color='gray')
plot_sensitivity_magic_performance()
plt.show()
```
image

### CODE REVIEW

File: lstchain/mc/plot_utils.py [hunk 1]
```diff
@@ -331,23 +336,22 @@ def plot_positions_survived_events(df_gammas,
     """
 
     e_reco_g = 10**df_gammas.mc_energy
-    e_reco_p = 10**df_protons.mc_energy
+    e_reco_p = 10 ** df_protons.mc_energy
```
Comment (maxnoe) [inline lstchain/mc/plot_utils.py:339]: Wat?

ereco = 10**mc_energy ?

Comment (vuillaut) [reply lstchain/mc/plot_utils.py:339]: I did not write that code but good catch.

Comment (vuillaut) [reply lstchain/mc/plot_utils.py]: This had nothing to do with this PR but I updated it nonetheless to take the reco energy from the dataframe

**Comment by moralejo** [comment]
This is stalled, but still needed, right?  - @vuillaut, can you solve the conflicts?

**Comment by vuillaut** [comment]
> This is stalled, but still needed, right? - @vuillaut, can you solve the conflicts?

Done

**Review by moralejo — APPROVED** [review]

**Comment by vuillaut** [comment]
Merged in #622

**Pull request closed without merging**